### PR TITLE
Attempt to patch up docker version and make consistent between ControlPlane and Nodes

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:
Attempt to patch up docker version and make consistent between ControlPlane and Nodes

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by detiber** [comment]
Completely untested as of right now.

File: cmd/clusterctl/examples/google/machine_setup_configs.yaml [hunk 1]
```diff
@@ -28,30 +28,32 @@ items:
...
-      touch /etc/apt/sources.list.d/kubernetes.list
-      sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list'
       apt-get update -y
-      apt-get install -y \
-        socat \
-        ebtables \
-        apt-transport-https \
-        cloud-utils \
-        prips
+      apt-get install -y socat ebtables apt-get install apt-transport-https ca-certificates curl software-properties-common cloud-utils prips
```
Comment (justinsb) [inline cmd/clusterctl/examples/google/machine_setup_configs.yaml]: Missing newline I think?

File: cmd/clusterctl/examples/google/machine_setup_configs.yaml [hunk 2]
```diff
@@ -28,30 +28,32 @@ items:
...
-        apt-get install -y docker.io
-        echo 'DOCKER_OPTS="--iptables=false --ip-masq=false"' > /etc/default/docker
-        systemctl daemon-reload
-        systemctl enable docker
-        systemctl start docker
+          # prevent docker from auto-starting
+          echo "exit 101" > /usr/sbin/policy-rc.d
+          chmod +x /usr/sbin/policy-rc.d
+          trap "rm /usr/sbin/policy-rc.d" RETURN
+          apt-get install -y docker-ce=18.06.2~ce~3-0~ubuntu
```
Comment (justinsb) [inline cmd/clusterctl/examples/google/machine_setup_configs.yaml]: I do wonder if we should just use the tar.gz packaging - the docker packages seem to cause as many problems as they solve.  We support it in kops, but only do it for OSes where docker doesn't package.  But when we go to containerd, I think we're going to try switching.

Comment (detiber) [reply cmd/clusterctl/examples/google/machine_setup_configs.yaml]: I was mainly trying to patch things up enough to get them working again without too much disruption :)

File: cmd/clusterctl/examples/google/machine_setup_configs.yaml (hunk 1 above)
Comment (detiber) [reply cmd/clusterctl/examples/google/machine_setup_configs.yaml]: Apparently `apt-get install` doesn't know how to install `apt-get` and `install`, fixed up in the latest version :joy:

**Comment by detiber** [comment]
@justinsb this should be good to go now. This doesn't completely fix up the `clusterctl create` workflow, but it does get further now.

There appears to be an issue during pivoting where the gcp provider controller manager fails to come up, but I don't currently have time to debug further.

**Comment by justinsb** [comment]
Going to merge this, as it's moving us in the right direction!

/approve
/lgtm

**Pull request merged**